packages/xstate-wallet/src/store/indexedDB-backend.ts

```diff
@@ -0,0 +1,249 @@
+import {BigNumber} from 'ethers/utils';
+import {MemoryChannelStoreEntry} from './memory-channel-storage';
+import {Objective, DBBackend} from './types';
+import 'fake-indexeddb/auto';
+
+enum ObjectStores {
+  channels = 'channels',
+  objectives = 'objectives',
+  nonces = 'nonces',
+  privateKeys = 'privateKeys',
+  ledgers = 'ledgers'
+}
+
+// A running, functioning example can be seen and played with here: https://codesandbox.io/s/elastic-kare-m1jp8
+export class IndexedDBBackend implements DBBackend {
+  private _db: any;
+
+  constructor() {
+    if (!indexedDB) {
+      console.error("Your browser doesn't support a stable version of IndexedDB.");
+    }
+  }
+  public async initialize() {
+    return new Promise<IDBDatabase>((resolve, reject) => {
+      const request = indexedDB.open('xstateWallet', 1);
+
+      request.onupgradeneeded = event => {
+        const db = (event.target as any).result;
+        db.createObjectStore('channels', {autoIncrement: true});
+        db.createObjectStore('objectives', {autoIncrement: true});
+        db.createObjectStore('nonces', {autoIncrement: true});
+        db.createObjectStore('privateKeys', {autoIncrement: true});
+        db.createObjectStore('ledgers', {autoIncrement: true});
+      };
+
+      request.onerror = err => reject(err);
+      request.onsuccess = () => {
+        this._db = request.result;
+        resolve(request.result);
+      };
+    });
+  }
+
+  // Generic Getters
+
+  public async channels() {
+    return this.getAll(ObjectStores.channels);
+  }
+  public async objectives() {
+    return this.getAll(ObjectStores.objectives);
+  }
+  public async nonces() {
+    return this.getAll(ObjectStores.nonces);
+  }
+  public async privateKeys() {
+    return this.getAll(ObjectStores.nonces);
+  }
+  public async ledgers() {
+    return this.getAll(ObjectStores.ledgers);
+  }
+
+  // Individual Getters
+
+  public async getChannel(key: string) {
+    return this.get(ObjectStores.channels, key);
+  }
+  public async getObjective(key: number) {
+    return this.get(ObjectStores.objectives, key);
+  }
+  public async getNonce(key: string) {
+    return this.get(ObjectStores.nonces, key);
+  }
+  public async getPrivateKey(key: string) {
+    return this.get(ObjectStores.privateKeys, key);
+  }
+  public async getLedger(key: string) {
+    return this.get(ObjectStores.ledgers, key);
+  }
+
+  // Individual Getters
+
+  public async setPrivateKey(key: string, value: string) {
+    return this.put(ObjectStores.privateKeys, value, key);
+  }
+  public async setChannel(key: string, value: MemoryChannelStoreEntry) {
+    return this.put(ObjectStores.channels, value, key);
+  }
```

Comment:
The "obsolete" store worked like:
- The store just stored plain data
- the ChannelStoreEntry was a class which accepted that data in its constructor
- `store.getEntry(channelId)` would construct a new class using the stored data.

The current MemoryChannelStoreEntry almost works like that. If you add:
```
  public data() {
    return {
      stateVariables: this.stateVariables,
      channelConstants: this.channelConstants,
      signatures: this.signatures,
      funding: this.funding,
      myIndex: this.myIndex
    };
  }

  static fromJson(data: string) {
    const {stateVariables, channelConstants, signatures, funding, myIndex} = JSON.parse(data);
    return new MemoryChannelStoreEntry(
      channelConstants,
      myIndex,
      stateVariables,
      signatures,
      funding
    );
  }
```
to the `MemoryChannelStoreEntry`, then you can do:
```suggestion
  public async setChannel(key: string, value: MemoryChannelStoreEntry) {
    return this.put(ObjectStores.channels, value.data, key);
  }
  
  public async getChannel(key: string): MemoryChannelStoreEntry {
    return MemoryChannelStoreEntry.fromJson(await this.get(ObjectStores.channels, key))
  }
```

Reply:
Thanks for that! Added